src/lib/migrations.test.ts

```diff
@@ -0,0 +1,293 @@
+import { beforeEach, test } from 'tap'
+import tmp from 'tmp'
+import crypto from 'crypto'
+import path from 'path'
+import fs from 'fs-extra'
+import Database, { Database as DatabaseInstance } from 'better-sqlite3'
+
+import { Migration, migrate } from './migrations'
+
+type TestContext = {
+  buildMigration: (name: string, query: string) => void
+  db: DatabaseInstance
+  getSQLiteTableInfo: (tableName?: string) => any[]
+  runMigrations: () => void
+}
+
+tmp.setGracefulCleanup()
+
+const TEST_TABLE_NAME = 'Developers'
+
+function generateMigrationChecksum(s: string) {
+  return crypto.createHash('sha256').update(s).digest('hex')
+}
+
+function migrationRecordedAsSuccess({
+  finished_at,
+  logs,
+  rolled_back_at,
+}: Migration) {
+  return finished_at !== null && logs === null && rolled_back_at === null
+}
+
+const fixtures = {
```

Comment:
pretty contrived examples but should sufficiently work for testing and make it more maintainable moving forward